app/controllers/api/v1/reservations_controller.rb

```diff
@@ -0,0 +1,38 @@
+class Api::V1::ReservationsController < ApplicationController
+  before_action :authenticate_user!
+  authorize_resource
+
+  def index
+    @reservations = current_user.reservations.all
+    render json: @reservations, status: :ok
+  end
```

Comment:
Great work on the reservations _controller! just a quick suggestion to optimize the index action. To address the `N+1` query problem, consider using includes with the Reservation model's association named doctor. This way, you can fetch all reservations for the current user and their associated doctors in a single query, enhancing database performance.
`  @reservations = current_user.reservations.includes(:doctor).all`